samza-core/src/main/java/org/apache/samza/system/inmemory/InMemorySystemFactory.java

```diff
@@ -31,20 +35,28 @@
  * Initial draft of in-memory {@link SystemFactory}. It is test only and not meant for production use right now.
  */
 public class InMemorySystemFactory implements SystemFactory {
-  private static final InMemoryManager MEMORY_MANAGER = new InMemoryManager();
+  private static Map<Integer, InMemoryManager> jobToManager = new HashMap<>();
```

Comment:
Does the access to `jobToManager` map should be thread-safe?

I think it would be better to synchronize all these factory methods.

Reply:
I suppose changing Map -> ConcurrentHashMap makes it thread safe, unless you have a better approach to recommend.

Reply:
Yes, that would work.

Do you know if `InMemoryManager` (and associated dependencies) are thread safe(Was unable to find it from documentation)?